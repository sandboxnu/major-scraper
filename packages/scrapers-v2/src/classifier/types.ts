export enum CatalogEntryType {
  Major = "Major",
  Minor = "Minor",
  Concentration = "Concentration",
  Unknown = "Unknown",
}

export type TypedCatalogEntry = { url: string; type: CatalogEntryType };
