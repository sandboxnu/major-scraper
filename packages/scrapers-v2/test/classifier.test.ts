import { classifyCatalogEntries } from "../src/classifier/classifier";
import {
  ACCELERATED_DEGREE_PROGRAM,
  ACCOUNTING_MINOR,
  AEROSPACE_MINOR,
  ARCHITECTURE_MINOR,
  BSCS,
  BUSINESS,
  CHEMICAL_ENG,
  CS_GAME_DEV,
  CS_HISTORY,
  CS_MUSIC_WITH_TECH_MAJOR,
  FINTECH_CONCENTRATION,
  MEDIA_SCREEN_STUDIES_HISTORY,
  WOMEN_GENDER_SEXUALITY_MINOR,
} from "./testUrls";

const inputs = [
  BSCS,
  CS_HISTORY,
  MEDIA_SCREEN_STUDIES_HISTORY,
  ACCOUNTING_MINOR,
  ARCHITECTURE_MINOR,
  FINTECH_CONCENTRATION,
  AEROSPACE_MINOR,
  ACCELERATED_DEGREE_PROGRAM,
  CS_MUSIC_WITH_TECH_MAJOR,
  WOMEN_GENDER_SEXUALITY_MINOR,
  BUSINESS,
  CHEMICAL_ENG,
  CS_GAME_DEV,
];

describe("Filters work", () => {
  for (const url of inputs) {
    test(url, async () => {
      expect(await classifyCatalogEntries([url])).toMatchSnapshot();
    });
  }
});
