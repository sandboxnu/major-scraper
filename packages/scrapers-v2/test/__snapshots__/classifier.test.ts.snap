// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters work https://catalog.northeastern.edu/undergraduate/arts-media-design/architecture/architectural-design-minor 1`] = `
Array [
  Object {
    "type": "Minor",
    "url": "https://catalog.northeastern.edu/undergraduate/arts-media-design/architecture/architectural-design-minor",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/arts-media-design/communication-studies/media-screen-studies-history-ba#programrequirementstext 1`] = `
Array [
  Object {
    "type": "Major",
    "url": "https://catalog.northeastern.edu/undergraduate/arts-media-design/communication-studies/media-screen-studies-history-ba#programrequirementstext",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/business/business-administration-bsba#programrequirementstext 1`] = `
Array [
  Object {
    "type": "Major",
    "url": "https://catalog.northeastern.edu/undergraduate/business/business-administration-bsba#programrequirementstext",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/business/concentrations/fintech 1`] = `
Array [
  Object {
    "type": "Concentration",
    "url": "https://catalog.northeastern.edu/undergraduate/business/concentrations/fintech",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/business/interdisciplinary-minors/accounting-advisory-services-minor 1`] = `
Array [
  Object {
    "type": "Minor",
    "url": "https://catalog.northeastern.edu/undergraduate/business/interdisciplinary-minors/accounting-advisory-services-minor",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/computer-information-science/accelerated-bachelor-graduate-degree-programs 1`] = `
Array [
  Object {
    "type": "Unknown",
    "url": "https://catalog.northeastern.edu/undergraduate/computer-information-science/accelerated-bachelor-graduate-degree-programs",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-information-science-combined-majors/computer-science-concentration-music-composition-technology-bs 1`] = `
Array [
  Object {
    "type": "Major",
    "url": "https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-information-science-combined-majors/computer-science-concentration-music-composition-technology-bs",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-information-science-combined-majors/computer-science-game-development-bs#programrequirementstext 1`] = `
Array [
  Object {
    "type": "Major",
    "url": "https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-information-science-combined-majors/computer-science-game-development-bs#programrequirementstext",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-information-science-combined-majors/computer-science-history-bs#programrequirementstext 1`] = `
Array [
  Object {
    "type": "Major",
    "url": "https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-information-science-combined-majors/computer-science-history-bs#programrequirementstext",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-science/bscs#programrequirementstext 1`] = `
Array [
  Object {
    "type": "Major",
    "url": "https://catalog.northeastern.edu/undergraduate/computer-information-science/computer-science/bscs#programrequirementstext",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/engineering/chemical/chemical-engineering-bsche#programrequirementstext 1`] = `
Array [
  Object {
    "type": "Major",
    "url": "https://catalog.northeastern.edu/undergraduate/engineering/chemical/chemical-engineering-bsche#programrequirementstext",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/engineering/mechanical-industrial/aerospace-minor 1`] = `
Array [
  Object {
    "type": "Minor",
    "url": "https://catalog.northeastern.edu/undergraduate/engineering/mechanical-industrial/aerospace-minor",
  },
]
`;

exports[`Filters work https://catalog.northeastern.edu/undergraduate/social-sciences-humanities/interdisciplinary/womens-gender-sexuality-studies-minor 1`] = `
Array [
  Object {
    "type": "Minor",
    "url": "https://catalog.northeastern.edu/undergraduate/social-sciences-humanities/interdisciplinary/womens-gender-sexuality-studies-minor",
  },
]
`;
